src/libraries/Fuzzing/DotnetFuzzing/Program.cs

```diff
@@ -0,0 +1,373 @@
+            .Where(t => t.GetInterfaces().Contains(typeof(IFuzzer)))
+            .Select(t => (IFuzzer)Activator.CreateInstance(t)!)
+            .OrderBy(f => f.Name, StringComparer.OrdinalIgnoreCase)
+            .ToArray();
+
+        void PrintUsage()
+        {
+            Console.Error.WriteLine(
+                $"""
+                Usage:
+                    DotnetFuzzing <Fuzzer name> [input file/directory]
+                    DotnetFuzzing prepare-onefuzz <output directory>
+
+                Fuzzers available: {string.Join(", ", fuzzers.Select(t => t.Name))}
+                """);
+        }
+
+        if (args.Length == 0)
+        {
+            PrintUsage();
+            return;
+        }
+
+        if (args[0].Equals("prepare-onefuzz", StringComparison.OrdinalIgnoreCase))
+        {
+            if (args.Length != 2)
+            {
+                PrintUsage();
+                return;
+            }
+
+            string publishDirectory = Path.GetDirectoryName(typeof(Program).Assembly.Location) ?? Environment.CurrentDirectory;
+
+            await PrepareOneFuzzDeploymentAsync(fuzzers, publishDirectory, args[1]);
+            return;
+        }
+
+        IFuzzer? fuzzer = fuzzers
+            .Where(f => f.Name.Equals(args[0], StringComparison.OrdinalIgnoreCase))
+            .FirstOrDefault();
+
+        if (fuzzer is null)
+        {
+            Console.Error.WriteLine($"Fuzzer '{args[0]}' not found. Available: {string.Join(", ", fuzzers.Select(t => t.Name))}");
+            return;
+        }
+
+        RunFuzzer(fuzzer, inputFiles: args.Length > 1 ? args[1] : null);
+    }
+
+    private static void RunFuzzer(IFuzzer fuzzer, string? inputFiles)
+    {
+        if (!string.IsNullOrEmpty(inputFiles))
+        {
+            string[] files = Directory.Exists(inputFiles)
+                ? Directory.GetFiles(inputFiles)
+                : [inputFiles];
+
+            foreach (string inputFile in files)
+            {
+                fuzzer.FuzzTarget(File.ReadAllBytes(inputFile));
+            }
+
+            return;
+        }
+
+        Fuzzer.LibFuzzer.Run(fuzzer.FuzzTarget);
+    }
+
+    private static async Task PrepareOneFuzzDeploymentAsync(IFuzzer[] fuzzers, string publishDirectory, string outputDirectory)
+    {
+        string[] dictionaries = Directory.GetFiles(Path.Combine(publishDirectory, "Dictionaries"))
+            .Select(Path.GetFileName)
+            .ToArray()!;
+
+        if (dictionaries.FirstOrDefault(dict => !fuzzers.Any(f => f.Dictionary == dict)) is { } unusedDictionary)
+        {
+            throw new Exception($"Dictionary '{unusedDictionary}' is not referenced by any fuzzer.");
+        }
+
+        Directory.CreateDirectory(outputDirectory);
+
+        await DownloadArtifactAsync(
+            Path.Combine(publishDirectory, "libfuzzer-dotnet.exe"),
+            "https://github.com/Metalnem/libfuzzer-dotnet/releases/download/v2023.06.26.1359/libfuzzer-dotnet-windows.exe",
```

Comment:
We're ok downloading such executables from an arbitrary repo?

Reply:
I don't know of a great alternative.

I figured that since the version is pinned (checksum), it's pretty much the same thing as us using the `SharpFuzz 2.1.1` NuGet package from the internal mirror / asking anyone working on fuzzers to install `dotnet tool install --global SharpFuzz.CommandLine`.